activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Improve `ActiveRecord::Store` to raise a descriptive exception if the column is not either
+    structured (e.g., PostgreSQL +hstore+/+json+, or MySQL +json+) or declared serializable via
+    `ActiveRecord.store`.
+
+    Previously, a `NoMethodError` would be raised when the accessor was read or written:
+
+        NoMethodError: undefined method `accessor' for an instance of ActiveRecord::Type::Text
+
+    Now, a descriptive `ConfigurationError` is raised:
+
+        ActiveRecord::ConfigurationError: the column 'metadata' has not been configured as a store.
```

Comment:
@flavorjones I believe the 'metadata' column is not meaningful here unless we define the metadata reference or use <store_attribute> to make it generic.

```ruby
  t.text :metadata
  store_accessor :metadata, [:likes_count, :comments_count, :tags]
```